src/java.base/share/classes/java/lang/Boolean.java

```diff
@@ -255,8 +255,8 @@ public static int hashCode(boolean value) {
      *          same value; {@code false} otherwise.
      */
     public boolean equals(Object obj) {
-        if (obj instanceof Boolean) {
-            return value == ((Boolean)obj).booleanValue();
+        if (obj instanceof Boolean b) {
+            return value == b.booleanValue();
```

Comment:
I would go even a step further and use `value` instead of `booleanValue()`, `intValue()` etc because there is no need to call a method to get the value.

```java
  if (obj instanceof Boolean b) {
            return value == b.value;
```

Reply:
Hi @forax,
Yes, I had considered that before sending out the PR. It shouldn't matter from a performance perspective, so it would come to down to which version of the code was marginally easier to read.

Reply:
Ah, too slow to answer ...
Anyway, the wrapper types tend to be used everywhere so there are also used in methods that are executed only a few times so only by the interpreter, in those case, accessing directly to the field is a win.